x-pack/plugins/security/public/nav_control/nav_control_component.tsx

```diff
@@ -32,174 +34,128 @@ export interface UserMenuLink {
-
-      // Set this as the first link if there is no user-defined profile link
-      if (!hasCustomProfileLinks) {
-        items.unshift(profileMenuItem);
-      } else {
-        items.push(profileMenuItem);
-      }
-    }
-
-    const logoutMenuItem = {
-      name: isAnonymousUser ? (
-        <FormattedMessage
-          id="xpack.security.navControlComponent.loginLinkText"
-          defaultMessage="Log in"
+export const SecurityNavControl: FunctionComponent<SecurityNavControlProps> = ({
+  editProfileUrl,
+  logoutUrl,
+  userMenuLinks$,
+}) => {
+  const userMenuLinks = useObservable(userMenuLinks$, []);
+  const [isOpen, setIsOpen] = useState(false);
+
+  const userProfile = useUserProfile<{ avatar: IUserAvatar }>('avatar');
+
+  const displayName = userProfile.value ? getUserDisplayName(userProfile.value.user) : '';
+
+  const button = (
+    <EuiHeaderSectionItemButton
+      aria-controls="headerUserMenu"
+      aria-expanded={isOpen}
+      aria-haspopup="true"
+      aria-label={i18n.translate('xpack.security.navControlComponent.accountMenuAriaLabel', {
+        defaultMessage: 'Account menu',
+      })}
+      onClick={() => setIsOpen((value) => (userProfile.value ? !value : false))}
+      data-test-subj="userMenuButton"
+      style={{ lineHeight: 'normal' }}
+    >
+      {userProfile.value ? (
+        <UserAvatar
+          user={userProfile.value.user}
+          avatar={userProfile.value.data.avatar}
+          size="s"
+          data-test-subj="userMenuAvatar"
         />
       ) : (
+        <EuiLoadingSpinner size="m" />
+      )}
+    </EuiHeaderSectionItemButton>
+  );
+
+  const isAnonymous = userProfile.value ? isUserAnonymous(userProfile.value.user) : false;
+  const items: EuiContextMenuPanelItemDescriptor[] = [];
+  if (userMenuLinks.length) {
+    const userMenuLinkMenuItems = userMenuLinks
+      .sort(({ order: orderA = Infinity }, { order: orderB = Infinity }) => orderA - orderB)
+      .map(({ label, iconType, href }: UserMenuLink) => ({
+        name: label,
+        icon: <EuiIcon type={iconType} size="m" />,
+        href,
+        'data-test-subj': `userMenuLink__${label}`,
+      }));
+    items.push(...userMenuLinkMenuItems);
+  }
+
+  if (!isAnonymous) {
+    const hasCustomProfileLinks = userMenuLinks.some(({ setAsProfile }) => setAsProfile === true);
+    const profileMenuItem = {
+      name: (
         <FormattedMessage
-          id="xpack.security.navControlComponent.logoutLinkText"
-          defaultMessage="Log out"
+          id="xpack.security.navControlComponent.editProfileLinkText"
+          defaultMessage="{profileOverridden, select, true{Preferences} other{Profile}}"
+          values={{ profileOverridden: hasCustomProfileLinks }}
         />
       ),
-      icon: <EuiIcon type="exit" size="m" />,
-      href: logoutUrl,
-      'data-test-subj': 'logoutLink',
+      icon: <EuiIcon type={hasCustomProfileLinks ? 'controlsHorizontal' : 'user'} size="m" />,
+      href: editProfileUrl,
```

Comment:
It's a bit more complicated than that. We'd need to make sure users can still open the link in a new tab akin to what `reactRouterNavigate` does. Let's fix this as a separate enhancement.

Reply:
Good point about middle-click, we wouldn't want to lose that functionality.

I took a look around and it appears that other consumers are handling this by specifying _both_ `href` and `onClick`, for example, [the "Add integrations" button on the Home page](https://github.com/elastic/kibana/blob/9d662b77bec867247421376ecb60a07202ed45f9/src/plugins/home/public/application/components/add_data/add_data.tsx#L67-L84). This supports middle-clicking but avoids full page reloads when left-clicking.
I noticed that's wrapped in a `RedirectAppLinks` component which may have something to do with it, not sure.

I'm fine with deferring this as a separate enhancement since the User Profile page already functions this way today. I opened an issue for it here: #130743